pkg/adscert/metrics/metrics.go

```diff
@@ -0,0 +1,155 @@
+const (
+	VerifyOutcomeTypeBody verifyOutcomeType = "body"
+	VerifyOutcomeTypeUrl  verifyOutcomeType = "url"
+)
+
+// adscertMetricsRegistry Prometheus registry capturing ads.cert related metrics
+var adscertMetricsRegistry *prometheus.Registry
+
+// Creates all metrics
+var (
+	DNSLookupCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "dns_lookup_count",
+		Help:      "The total number of requests verified",
+	}, []string{dnsLookupErrorLabel})
+	DNSLookupTimeHistogram = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Name:      "dns_lookup_ms",
+		Help:      "Milliseconds to lookup DNS.",
+		Buckets:   standardMillisecondBuckets,
+	})
+	SignCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "sign_count",
+		Help:      "The total number of requests signed.",
+	}, []string{signErrorLabel})
+	SignTimeHistogram = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Name:      "sign_time_us",
+		Help:      "Microseconds to sign a request.",
+		Buckets:   standardMicrosecondBuckets,
+	})
+	VerifyCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "verify_count",
+		Help:      "The total number of requests verified.",
+	}, []string{verifyErrorLabel})
+	VerifyOutcomeCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "verify_outcome_count",
+		Help:      "The total number of valid signed requests.",
+	}, []string{verifyOutcomeTypeLabel, verifyOutcomeValidLabel})
+	VerifyTimeHistogram = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Name:      "verify_time_us",
+		Help:      "Microseconds to verify a request.",
+		Buckets:   standardMicrosecondBuckets,
+	})
+)
+
+// All metric collectors
+var collectors []prometheus.Collector = []prometheus.Collector{
+	DNSLookupCounter,
+	DNSLookupTimeHistogram,
+	SignCounter,
+	SignTimeHistogram,
+	VerifyCounter,
+	VerifyOutcomeCounter,
+	VerifyTimeHistogram,
+}
+
+func init() {
+	adscertMetricsRegistry = prometheus.NewRegistry()
+	for _, collector := range collectors {
+		adscertMetricsRegistry.MustRegister(collector)
+	}
+}
+
+func GetAdscertMetricsRegistry() *prometheus.Registry {
+	return adscertMetricsRegistry
+}
+
+func RecordDNSLookup(err adscerterrors.DNSLookupErrorCode) {
+	var DNSLookupErr string
+
+	if err != nil {
+		DNSLookupErr = err.Code
+	}
+
+	DNSLookupCounter.With(prometheus.Labels{
+		dnsLookupErrorLabel: DNSLookupErr,
+	}).Inc()
+}
+
+func RecordDNSLookupTime(observeTimeMilliseconds int64) {
+	DNSLookupTimeHistogram.Observe(float64(observeTimeMilliseconds))
+}
+
+func RecordSigning(err adscerterrors.SigningErrorCode) {
+	var signError string
+
+	if err != nil {
+		signError = err.Code
+	}
+
+	SignCounter.With(prometheus.Labels{
+		signErrorLabel: signError,
+	}).Inc()
+}
+
+func RecordSigningTime(observeTimeMicroseconds int64) {
```

Comment:
nit: please use time.Duration as function arguments for this and other functions that accept a time duration value.  You can hide all of the details about what time granularity gets recorded within this implementation.